streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -906,12 +908,16 @@ abstract class DStream[T: ClassTag] (
   /**
    * Save each RDD in this DStream as at text file, using string representation
    * of elements. The file name at each batch interval is generated based on
-   * `prefix` and `suffix`: "prefix-TIME_IN_MS.suffix".
+   * `prefix` and `suffix`: "prefix-TIME_IN_MS.suffix". If the `CompressionCodec`
+   * is defined, it will use specific `CompressionCodec` to compress the text.
    */
-  def saveAsTextFiles(prefix: String, suffix: String = ""): Unit = ssc.withScope {
```

Comment:
Yeah I'm surprised because this should be adding a parameter in the bytecode, but I could be missing something. Yes, I thought the general idea was not to add more methods to `DStream` as in retrospect they were generally not adding enough value over operating on `RDD`s.